code/modules/surgery/organs/heart.dm

```diff
@@ -167,21 +168,26 @@
 
 /obj/item/organ/heart/cybernetic/on_life()
 	. = ..()
-	if(dose_available && owner.stat == UNCONSCIOUS && !owner.reagents.has_reagent(rid))
-		owner.reagents.add_reagent(rid, ramount)
+	if(dose_available && owner.health <= owner.crit_threshold && !owner.reagents.has_reagent(reagid[1]))
+		to_chat(owner, "<span class='notice'>You hear a soft beep from your chest as your cybernetic heart activates.</span>")
 		used_dose()
 
 /obj/item/organ/heart/cybernetic/proc/used_dose()
 	dose_available = FALSE
+	owner.reagents.add_reagent(reagid[1], reagamount[1])
```

Comment:
I think it would be better if the base `used_dose` always went through the entire list, with the non-upgraded heart having a smaller list than the upgraded heart.